lib/draw/BaseRenderer.js

```diff
@@ -25,13 +25,17 @@ export default function BaseRenderer(eventBus, renderPriority) {
     }
   });
 
-  eventBus.on([ 'render.getShapePath', 'render.getConnectionPath'], renderPriority, function(evt, element) {
+  eventBus.on('render.getShapePath', renderPriority, function(event, element) {
     if (self.canRender(element)) {
-      if (evt.type === 'render.getShapePath') {
-        return self.getShapePath(element);
-      } else {
-        return self.getConnectionPath(element);
-      }
+      return self.getShapePath(element);
+    }
+  });
+
+  eventBus.on('render.getConnectionPath', renderPriority, function(event, connectionOrWaypoints) {
+    if ((connectionOrWaypoints.waypoints && self.canRender(connectionOrWaypoints)) ||
+        !connectionOrWaypoints.waypoints) {
```

Comment:
This will render regardless of the `canRender` result if the connection has no waypoints. Is this intentional?

Reply:
It will try to get the connection path for non-existing waypoints. That's an issue. The reason I changed this is that I want to be able to crop waypoints without needing the connection itself. Maybe the approach ist wrong.

Reply:
The point here is: `self.canRender(...)` is specifically there to decide, if the renderer should be turned to to get the connection path.

If the connection does not have waypoints the renderer will always be turned to, even it it reported _I CANNOT RENDER THIS CONNECTION_ via `self.canRender(...)`.

Reply:
I've removed this commit and went for a much simpler approach that I used initially. Instead of being able to pass waypoints instead of a connection I simply did

```
var croppedWaypoints = connectionDocking.getCroppedWaypoints(
      assign({}, connection, {
        waypoints: waypoints
      }),
      source,
      target
    );
```

in dmn-js.